subsys/usb/class/cdc_acm.c

```diff
@@ -625,34 +607,20 @@ static int cdc_acm_fifo_fill(struct device *dev,
  *
  * @return Number of bytes read.
  */
-static int cdc_acm_fifo_read(struct device *dev, u8_t *rx_data,
-			     const int size)
+static int cdc_acm_fifo_read(struct device *dev, u8_t *rx_data, const int size)
 {
-	u32_t avail_data, bytes_read, i;
 	struct cdc_acm_dev_data_t * const dev_data = DEV_DATA(dev);
+	u32_t len;
 
-	avail_data = (CDC_ACM_BUFFER_SIZE + dev_data->rx_buf_head -
-		      dev_data->rx_buf_tail) % CDC_ACM_BUFFER_SIZE;
-	if (avail_data > size) {
-		bytes_read = size;
-	} else {
-		bytes_read = avail_data;
-	}
-
-	for (i = 0U; i < bytes_read; i++) {
-		rx_data[i] = dev_data->rx_buf[(dev_data->rx_buf_tail + i) %
-					      CDC_ACM_BUFFER_SIZE];
-	}
+	len = MIN(size, ring_buf_space_get(dev_data->rx_ringbuf));
```

Comment:
Should this rather be something like
`len = MIN(size, dev_data->rx_ringbuf->size);`
or 
`len = MIN(size, dev_data->rx_ringbuf->size - ring_buf_space_get(dev_data->rx_ringbuf));`

Currently, no data will be read out using `uart_fifo_read(...)`if the ring buffer is full, as `ring_buf_space_get(dev_data->rx_ringbuf)` returns 0 free space left in the ring buffer.

Reply:
Thanks for noticing this, I will fix it.
probably we are not supposed to use `rx_ringbuf->size` directly, and size should be multiplied by 4 since `Size of buf in 32-bit chunks`. Maybe we can just use `size` directly.

Reply:
That's true. I think you're right in that using `size` directly is fine.